### ISSUE TITLE
`check_docs_formatted.py` should allow examples for PYI rules to be formatted like stub files

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ruff#11568]
Currently the script complains if you try to apply `.pyi`-style formatting to any examples. But many of the PYI rules are specific to stubs, so it makes sense to format the examples for these rules as though they are stubs.

**Comment by AlexWaygood** [comment]
> Could we extend our parsing to support markdown language specifier
> 
> ```
> ```pyi
> stub_file
> ```
> ```

That's a really interesting idea. I've never seen anybody use `pyi` as a markdown language specifier anywhere else, as in terms of syntax highlighting `.pyi` files obey the exact same rules as `.py` files. But for us it makes total sense, as we want to differentiate between the two in terms of formatting style.

**Comment by charliermarsh** [comment]
What does MkDocs / GitHub do if you use `pyi` as the specifier? I assume it's fine?

**Comment by Avasam** [comment]
> What does GitHub do if you use pyi as the specifier?

Best way to know is to try!

```py
# py
class A:
  def foo(bar: None = None): pass
  ...
```
```pyi
# pyi
class A:
  def foo(bar: None = None): pass
  ...
```

**Comment by charliermarsh** [comment]
Interesting. Discord at least doesn't get it:

![Screenshot 2024-05-28 at 10 04 27 PM](https://github.com/astral-sh/ruff/assets/1309177/bb79e3db-1f2b-4e8a-969f-d54deac42212)

**Comment by Avasam** [comment]
Discord uses https://github.com/highlightjs/highlight.js can always make a request to add `pyi` to the Python aliases

**Comment by calumy** [comment]
Closed by #13116

**Comment by zanieb** [comment]
Thanks!
